scripts/benchmark/sampling/sampling_visualization.py

```diff
@@ -0,0 +1,235 @@
+
+    left_column, right_column = st.columns(2)
+
+    with left_column:
+        st.metric('Total number of projects:', len(row_data), help='The number of projects in the dataset.')
+
+    with right_column:
+        st.metric(
+            'Actual number of projects:',
+            len(data),
+            delta=len(data) - len(row_data),
+            help='The number of projects in which all the specified metrics are present.',
+        )
+
+    st.header('Histograms')
+
+    metric_bins = {}
+    for metric in metrics:
+        st.subheader(metric)
+
+        metric_bins[metric] = get_bins(metric)
+        values, bin_edges = np.histogram(data[metric], bins=metric_bins[metric])
+
+        number_of_bins = len(bin_edges) - 1
+        number_of_non_empty_bins = len([value for value in values if value != 0])
+
+        left_column, right_column = st.columns(2)
+
+        with left_column:
+            st.metric('Number of bins:', number_of_bins)
+
+        with right_column:
+            st.metric(
+                'Number of non-empty bins:',
+                number_of_non_empty_bins,
+                delta=number_of_non_empty_bins - number_of_bins,
+            )
+
+        with st.expander('Histogram:'):
+            fig = px.bar(x=get_middle_of_bins(bin_edges), y=values)
+            fig.update_layout(xaxis_title='Bin', yaxis_title='Number of projects')
+            st.plotly_chart(fig, use_container_width=True)
+
+    st.header('Sampling')
+
+    converted_data = convert_to_intervals(data, metric_bins)
+    st.metric(
+        'Number of non-empty groups:',
+        len(converted_data.groupby(metrics, observed=True)),
+        help='Number of non-empty groups, after grouping by all metrics.',
+    )
+
+    left_column, right_column = st.columns(2)
+
+    with left_column:
+        sample_size = st.number_input(
+            'Sample size:',
+            min_value=1,
+            value=round(len(converted_data) * 0.2),
+            help='The expected number of projects in the sample.',
+        )
+
+    with right_column:
+        random_state = st.number_input('Random state:', value=42, help='Seed for random number generator.')
+
+    if st.button('Sample'):
+        with st.spinner('Sampling...'):
+            sample = get_stratified_sample(converted_data, metrics, sample_size, random_state)
+
+        st.metric(
+            'Actual sample size:',
+            len(sample),
+            delta=len(sample) - sample_size,
+            help=(
+                'Since it is impossible to collect projects from some small groups with saving the distribution, '
+                'the actual number of projects in the sample may be less than expected. '
+            ),
+        )
+
+        for metric in metrics:
+            with st.expander(metric):
+                compare_histograms(data, sample, metric, metric_bins[metric])
+
+        st.download_button('Download', sample[PROJECT_COLUMN].to_csv(index=False), 'sample.csv', 'text/csv')
```

Comment:
Should we add some information about it to the README file?

Reply:
Because currently, it says only about visualization, not about downloading samples from the web page.

Reply:
Also, it would be great to add a button to make and download yaml config by the chosen values, what do you think?